spec/integrations/consumption/non_critical_error_exponential_backoff.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# When we have non-critical error happening couple times and we use exponential backoff, Karafka
+# should increase the backoff time with each occurrence until max backoff.
+
+setup_karafka do |config|
+  config.max_wait_time = 100
+  config.max_messages = 1
+  config.pause_with_exponential_backoff = true
+  config.pause_max_timeout = 5_000
+  config.pause_timeout = 200
+end
+
+class Consumer < Karafka::BaseConsumer
+  def initialize
+    super
+    DataCollector.data[0] << Time.now.to_f
+  end
+
+  def consume
+    DataCollector.data[0] << Time.now.to_f
+
+    raise StandardError
+  end
+end
+
+draw_routes(Consumer)
+
+produce(DataCollector.topic, '0')
+
+start_karafka_and_wait_until do
+  DataCollector.data[0].size >= 10
+end
+
+# Backoff time before next exception occurrence (not before resume). Because of that and the fact
+# that we run this in parallel, we add some extra time to compensate.
+BACKOFF_RANGES = [
+  0..0.5,
+  0..1,
```

Comment:
Why do we need 0..1 three times?

Reply:
Because this is how the backoff algorithm works with smaller ranges.